TestBenches/MatchProcessorL3_test.cpp

```diff
@@ -0,0 +1,158 @@
+  if (not openDataFile(fin_tproj[1], "MP_L3PHIC/TrackletProjections_TPROJ_L1L2F_L3PHIC_04.dat")) return -1;
+  if (not openDataFile(fin_tproj[2], "MP_L3PHIC/TrackletProjections_TPROJ_L1L2G_L3PHIC_04.dat")) return -1;
+  if (not openDataFile(fin_tproj[3], "MP_L3PHIC/TrackletProjections_TPROJ_L1L2H_L3PHIC_04.dat")) return -1;
+  if (not openDataFile(fin_tproj[4], "MP_L3PHIC/TrackletProjections_TPROJ_L1L2I_L3PHIC_04.dat")) return -1;
+  if (not openDataFile(fin_tproj[5], "MP_L3PHIC/TrackletProjections_TPROJ_L1L2J_L3PHIC_04.dat")) return -1;
+  if (not openDataFile(fin_tproj[6], "MP_L3PHIC/TrackletProjections_TPROJ_L1L2K_L3PHIC_04.dat")) return -1;
+  if (not openDataFile(fin_tproj[7], "MP_L3PHIC/TrackletProjections_TPROJ_L5L6B_L3PHIC_04.dat")) return -1;
+  if (not openDataFile(fin_tproj[8], "MP_L3PHIC/TrackletProjections_TPROJ_L5L6C_L3PHIC_04.dat")) return -1;
+  if (not openDataFile(fin_tproj[9], "MP_L3PHIC/TrackletProjections_TPROJ_L5L6D_L3PHIC_04.dat")) return -1;
+
+  ifstream fin_vmstub;
+
+  bool validvmstub = openDataFile(fin_vmstub,"MP_L3PHIC/VMStubs_VMSME_L3PHICn1_04.dat");
+  if (not validvmstub) return -1;
+
+  if (not openDataFile(fin_as,"MP_L3PHIC/AllStubs_AS_L3PHICn1_04.dat")) return -1;
+
+  // open file(s) with reference results
+  ifstream fout_fm1;
+  ifstream fout_fm2;
+  ifstream fout_fm3;
+  ifstream fout_fm4;
+  ifstream fout_fm5;
+  ifstream fout_fm6;
+  ifstream fout_fm7;
+
+  if (not openDataFile(fout_fm1,"MP_L3PHIC/FullMatches_FM_L1L2_L3PHIC_04.dat")) return -1;
+  if (not openDataFile(fout_fm3,"MP_L3PHIC/FullMatches_FM_L5L6_L3PHIC_04.dat")) return -1;
+
+  // loop over events
+  for (int ievt = 0; ievt < nevents; ++ievt) {
+    cout << "Event: " << dec << ievt << endl;
+
+    fullmatch[0].clear();
+//    fullmatch[1].clear();
+//    fullmatch[2].clear();
+    fullmatch[3].clear();
+//    fullmatch[4].clear();
+//    fullmatch[5].clear();
+//    fullmatch[6].clear();
+//    fullmatch[7].clear();
+
+    // read event and write to memories
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj[0], fin_tproj[0], ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj[1], fin_tproj[1], ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj[2], fin_tproj[2], ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj[3], fin_tproj[3], ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj[4], fin_tproj[4], ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj[5], fin_tproj[5], ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj[6], fin_tproj[6], ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj[7], fin_tproj[7], ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj[8], fin_tproj[8], ievt);
+    writeMemFromFile<TrackletProjectionMemory<BARRELPS> >(tproj[9], fin_tproj[9], ievt);
```

Comment:
One last thing (sorry!), could this be put in a loop? :-)

Reply:
Good point, I just pushed this change.